src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -116,14 +132,33 @@ public StaticLangApiMethodView generateUnpagedListCallableMethod(
     return methodViewBuilder.type(ApiMethodType.UnpagedListCallableMethod).build();
   }
 
+  public StaticLangApiMethodView generateFlattenedAsyncMethod(
+      MethodTransformerContext context,
+      ImmutableList<Field> fields,
+      List<ParamWithSimpleDoc> additionalParams,
+      ApiMethodType type) {
+    StaticLangApiMethodView.Builder methodViewBuilder = StaticLangApiMethodView.newBuilder();
+
+    setCommonFields(context, methodViewBuilder);
+    methodViewBuilder.name(context.getNamer().getAsyncApiMethodName(context.getMethod()));
+    methodViewBuilder.callableName(context.getNamer().getCallableName(context.getMethod()));
+    setFlattenedMethodFields(context, fields, additionalParams, true, methodViewBuilder);
+    setStaticLangReturnFields(context, true, methodViewBuilder);
+
+    return methodViewBuilder.type(type).build();
+  }
+
   public StaticLangApiMethodView generateFlattenedMethod(
-      MethodTransformerContext context, ImmutableList<Field> fields) {
+      MethodTransformerContext context,
+      ImmutableList<Field> fields,
+      List<ParamWithSimpleDoc> additionalParams) {
     StaticLangApiMethodView.Builder methodViewBuilder = StaticLangApiMethodView.newBuilder();
 
     setCommonFields(context, methodViewBuilder);
     methodViewBuilder.name(context.getNamer().getApiMethodName(context.getMethod()));
-    setFlattenedMethodFields(context, fields, methodViewBuilder);
-    setStaticLangSyncReturnFields(context, methodViewBuilder);
+    methodViewBuilder.callableName(context.getNamer().getCallableName(context.getMethod()));
+    setFlattenedMethodFields(context, fields, additionalParams, false, methodViewBuilder);
+    setStaticLangReturnFields(context, false, methodViewBuilder);
```

Comment:
I can't tell what the "false" values are for here. Could you make a boolean that captures the notion of Sync/Async?